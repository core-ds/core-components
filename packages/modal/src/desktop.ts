export * from './Component.desktop';
export { ModalDesktopProps } from './typings';
