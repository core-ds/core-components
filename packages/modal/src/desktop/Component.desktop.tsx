import React, { forwardRef } from 'react';

import { Modal } from '../Component';
import { Content } from '../components/content/Component';
import { Controls, ControlsProps } from '../components/controls';
import { Footer } from '../components/footer/Component';
import { Header, HeaderProps } from '../components/header/Component';
import type { ModalDesktopProps } from '../typings';

const ModalDesktopComponent = forwardRef<HTMLDivElement, ModalDesktopProps>((props, ref) => (
    <Modal {...props} ref={ref} view='desktop' />
));

export const ModalDesktop = Object.assign(ModalDesktopComponent, {
    Content,
    Header: Header as React.FC<Omit<HeaderProps, 'titleSize' | 'subtitle'>>,
    Footer,
    Controls: Controls as React.FC<Omit<ControlsProps, 'mobileLayout'>>,
});
