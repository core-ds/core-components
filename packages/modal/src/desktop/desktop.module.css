@import '../vars.css';

.wrapper {
    padding-top: var(--modal-vertical-padding);
    padding-bottom: var(--modal-vertical-padding);
}

.component {
    width: 100%;
    max-width: 100%;
    border-radius: var(--modal-border-radius);
}

.fullscreen {
    padding-top: var(--gap-0);
    padding-bottom: var(--gap-0);
    overflow: hidden;

    & .component {
        flex: 1;
        border-radius: 0;
        overflow: auto;
    }
}

.size-500 {
    width: var(--modal-s-width);
}

.size-600 {
    width: var(--modal-m-width);
}

.size-800 {
    width: var(--modal-l-width);
}

.size-1140 {
    width: var(--modal-xl-width);
}
