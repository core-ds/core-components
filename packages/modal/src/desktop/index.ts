export * from './Component.desktop';
export { type ModalDesktopProps } from '../typings';
