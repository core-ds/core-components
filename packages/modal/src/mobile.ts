export * from './Component.mobile';
export { ModalMobileProps } from './typings';
