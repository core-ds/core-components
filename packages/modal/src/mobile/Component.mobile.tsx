import React, { forwardRef } from 'react';

import { Modal } from '../Component';
import { Content } from '../components/content/Component';
import { Controls, ControlsProps } from '../components/controls';
import { Footer } from '../components/footer/Component';
import { Header } from '../components/header/Component';
import type { ModalMobileProps } from '../typings';

const ModalMobileComponent = forwardRef<HTMLDivElement, ModalMobileProps>((props, ref) => (
    <Modal {...props} ref={ref} view='mobile' />
));

const ControlsMobile = Controls as React.FC<Omit<ControlsProps, 'mobileLayout'>>;

export const ModalMobile = Object.assign(ModalMobileComponent, {
    Content,
    Header,
    Footer,
    Controls: ControlsMobile,
});

export {
    ModalMobileComponent as ModalComponentMobile,
    Content as ContentMobile,
    Header as HeaderMobile,
    Footer as FooterMobile,
    ControlsMobile,
};
