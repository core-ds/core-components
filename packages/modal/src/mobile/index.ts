export * from './Component.mobile';
export { type ModalMobileProps } from '../typings';
