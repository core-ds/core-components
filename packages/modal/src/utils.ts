import { getDataTestId } from '@alfalab/core-components-shared';

import { type ModalResponsiveProps } from './typings';

export function getModalTestIds(dataTestId: string) {
    return {
        modal: dataTestId,
        content: getDataTestId(dataTestId, 'content'),
        footer: getDataTestId(dataTestId, 'footer'),
        controls: getDataTestId(dataTestId, 'controls'),
        header: getDataTestId(dataTestId, 'header'),
        title: getDataTestId(dataTestId, 'header-title'),
        closer: getDataTestId(dataTestId, 'header-closer'),
        backButton: getDataTestId(dataTestId, 'header-back-button'),
    };
}

export const getSizeStyle = (size: ModalResponsiveProps['size']) => {
    if (size === 'fullscreen') {
        return size;
    }

    return `size-${size}`;
};
