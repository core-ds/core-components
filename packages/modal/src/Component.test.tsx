import React from 'react';
import { ModalDesktop, ModalDesktopProps } from './desktop';
import { ModalMobile, ModalMobileProps } from './mobile';
import { render, screen } from '@testing-library/react';
import { getModalTestIds } from './utils';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

const ModalDesktopWrapper = (props: Partial<ModalDesktopProps>) => {
    return (
        <ModalDesktop open={true} transitionProps={{ timeout: 0 }} {...props}>
            <ModalDesktop.Header hasCloser={true} hasBackButton={true} title='Title' />
            <ModalDesktop.Content>Content</ModalDesktop.Content>
            <ModalDesktop.Footer>
                <ModalDesktop.Controls primary='Primary' secondary='Secondary' />
            </ModalDesktop.Footer>
        </ModalDesktop>
    );
};

const ModalMobileWrapper = (props: Partial<ModalMobileProps>) => {
    return (
        <ModalMobile open={true} {...props}>
            <ModalMobile.Header hasCloser={true} hasBackButton={true} title='Title' />
            <ModalMobile.Content>Content</ModalMobile.Content>
            <ModalMobile.Footer>
                <ModalMobile.Controls primary='Primary' secondary='Secondary' />
            </ModalMobile.Footer>
        </ModalMobile>
    );
};

const COMPONENT_NAME_TO_WRAPPER = {
    ModalDesktop: ModalDesktopWrapper,
    ModalMobile: ModalMobileWrapper,
} as const;

(['ModalDesktop', 'ModalMobile'] as const).forEach((componentName) => {
    const Component = COMPONENT_NAME_TO_WRAPPER[componentName];

    describe(componentName, () => {
        describe('snapshots tests', () => {
            it('should match snapshot', () => {
                render(<Component />);
                expect(screen.getByRole('dialog')).toMatchSnapshot();
            });
        });

        describe('attributes test', () => {
            it('should have data-test-id', () => {
                const dti = 'modal-dti';
                const { getByTestId } = render(<Component dataTestId={dti} />);

                const testIds = getModalTestIds(dti);

                expect(getByTestId(testIds.modal)).toBeInTheDocument();
                expect(getByTestId(testIds.header)).toBeInTheDocument();
                expect(getByTestId(testIds.title)).toBeInTheDocument();
                expect(getByTestId(testIds.content)).toBeInTheDocument();
                expect(getByTestId(testIds.footer)).toBeInTheDocument();
                expect(getByTestId(testIds.controls)).toBeInTheDocument();
                expect(getByTestId(testIds.closer)).toBeInTheDocument();
                expect(getByTestId(testIds.backButton)).toBeInTheDocument();
            });
        });
    });
});
