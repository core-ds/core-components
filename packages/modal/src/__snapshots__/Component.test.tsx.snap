// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDesktop snapshots tests should match snapshot 1`] = `
<div
  class="wrapper wrapper"
  role="dialog"
  style="z-index: 100;"
  tabindex="-1"
>
  <div
    class="component size-500 appear appearActive"
  >
    <div
      class="content"
    >
      <div
        class="header hasContent size-500"
      >
        <div
          class="mainLine"
        >
          <div
            class="addonsWrapper"
          >
            <div
              class="addon"
            >
              <button
                aria-label="назад"
                class="component text size-48 hug size-48 component text component text component"
                type="button"
              >
                <span
                  class="label stretchText"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="iconWrapper"
                    >
                      <svg
                        fill="currentColor"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M1 12l8.25-9 2.25 2.25-5.25 5.25H23v3H6.25l5.25 5.25L9.25 21 1 12z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text primary-large medium"
                      style="opacity: 1;"
                    >
                      Назад
                    </span>
                  </div>
                </span>
              </button>
            </div>
          </div>
          <div
            class="addonsWrapper rightAddons"
          >
            <div
              class="addon closer"
            >
              <div
                class="closer"
              >
                <button
                  aria-label="закрыть"
                  class="component text size-48 hug size-48 component text component text cc-icon-button button primary component border-40"
                  type="button"
                >
                  <span
                    class="label stretchText"
                  >
                    <span
                      class="iconWrapper size-48 center"
                    >
                      <svg
                        class="icon"
                        fill="currentColor"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M10 12L3 5.5 5.25 3 12 9.5 18.75 3 21 5.5 14 12l7 6.5-2.25 2.5L12 14.5 5.25 21 3 18.5l7-6.5z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="content contentOnBot left trim"
          style="visibility: visible;"
        >
          <div
            class="title"
          >
            Title
          </div>
        </div>
      </div>
      <div
        class="content withHeader size-500"
      >
        Content
      </div>
      <div
        class="footer start size-500"
      >
        <div
          class="component start"
        >
          Primary
          Secondary
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalMobile snapshots tests should match snapshot 1`] = `
<div
  class="wrapper"
  role="dialog"
  style="z-index: 100;"
  tabindex="-1"
>
  <div
    class="component appear appearActive"
  >
    <div
      class="content"
    >
      <div
        class="header hasContent header"
      >
        <div
          class="mainLine"
        >
          <div
            class="addonsWrapper"
          >
            <div
              class="addon"
            >
              <button
                aria-label="назад"
                class="component text size-32 hug size-32 component text component text component"
                type="button"
              >
                <span
                  class="label stretchText"
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="iconWrapper"
                    >
                      <svg
                        fill="currentColor"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M7.5 13H19v-2H7.5L11 7.5 9.5 6 4 12l5.5 6 1.5-1.5L7.5 13z"
                        />
                      </svg>
                    </div>
                    <span
                      class="text component medium"
                      style="opacity: 1;"
                    >
                      Назад
                    </span>
                  </div>
                </span>
              </button>
            </div>
          </div>
          <div
            class="addonsWrapper rightAddons"
          >
            <div
              class="addon closer"
            >
              <div
                class="closer"
              >
                <button
                  aria-label="закрыть"
                  class="component text size-48 hug size-48 component text component text cc-icon-button button primary component border-32"
                  type="button"
                >
                  <span
                    class="label stretchText"
                  >
                    <span
                      class="iconWrapper size-32 center"
                    >
                      <svg
                        class="icon"
                        fill="currentColor"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                      >
                        <path
                          d="M10.586 12L6 16.5 7.5 18l4.5-4.586L16.5 18l1.5-1.5-4.586-4.5L18 7.5 16.5 6 12 10.586 7.5 6 6 7.5l4.586 4.5z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="content contentOnBot content left trim"
          style="visibility: visible;"
        >
          <div
            class="title"
          >
            Title
          </div>
        </div>
      </div>
      <div
        class="content withHeader content"
      >
        Content
      </div>
      <div
        class="footer start footer"
      >
        <div
          class="component start"
        >
          Secondary
          Primary
        </div>
      </div>
    </div>
  </div>
</div>
`;
